specification/purview/resource-manager/Microsoft.Purview/stable/2021-12-01/purview.json

```diff
@@ -0,0 +1,2685 @@
+          "$ref": "#/definitions/AccountProperties",
+          "description": "The account properties."
+        },
+        "tags": {
+          "description": "Tags on the azure resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AccessKeys": {
+      "description": "The Account access keys.",
+      "type": "object",
+      "properties": {
+        "atlasKafkaPrimaryEndpoint": {
+          "description": "Gets or sets the primary connection string.",
+          "type": "string"
+        },
+        "atlasKafkaSecondaryEndpoint": {
+          "description": "Gets or sets the secondary connection string.",
+          "type": "string"
+        }
+      }
+    },
+    "CollectionAdminUpdate": {
+      "description": "Collection administrator update.",
+      "type": "object",
+      "properties": {
+        "objectId": {
+          "description": "Gets or sets the object identifier of the admin.",
+          "type": "string"
+        }
+      }
+    },
+    "DefaultAccountPayload": {
+      "description": "Payload to get and set the default account in the given scope",
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "description": "The name of the account that is set as the default.",
+          "type": "string"
+        },
+        "resourceGroupName": {
+          "description": "The resource group name of the account that is set as the default.",
+          "type": "string"
+        },
+        "scope": {
+          "description": "The scope object ID. For example, sub ID or tenant ID.",
+          "type": "string"
+        },
+        "scopeTenantId": {
+          "description": "The scope tenant in which the default account is set.",
+          "type": "string"
+        },
+        "scopeType": {
+          "description": "The scope where the default account is set.",
+          "enum": [
+            "Tenant",
+            "Subscription"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "scopeType",
+            "modelAsString": true
+          }
+        },
+        "subscriptionId": {
+          "description": "The subscription ID of the account that is set as the default.",
+          "type": "string"
+        }
+      }
+    },
+    "BatchFeatureRequest": {
+      "description": "Feature request model",
+      "type": "object",
+      "properties": {
+        "features": {
+          "description": "Set of features",
+          "uniqueItems": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "BatchFeatureStatus": {
+      "description": "List of features with enabled status",
+      "type": "object",
+      "properties": {
+        "features": {
+          "description": "Features with enabled status",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
ARM recommends against the use of the additionalProperties/empty/unschematized object pattern. ARM favors stronger typing wherever possible because it allows all consumers of your API to reason about it. This has benefits for everyone that interacts with your API including basic infrastructure, other Azure features, SDK owners, and customers. Unschematized objects are essentially API contracts that bypass Azure versioning infrastructure.

For example, customers will not be able to manage these objects using Azure Policy, or query them using ARG, both of which are standard Azure features. Customers expect to be able to use all common Azure features against all of their resources.

Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined.

Reply:
features property is Dictionary<string, bool>.
What would be the recommendation here?

Reply:
Had offline discussion - 
Keys are the ones that customer specifies in the request payload
Basically size of the dictionary is dynamic